lib/puppet/provider/registry_value/registry.rb

```diff
@@ -201,7 +192,7 @@ def data_to_bytes(type, data)
   end
 
   def write(reg, name, type, data)
-    from_string_to_wide_string(valuename) do |name_ptr|
+    FFI::MemoryPointer.from_string_to_wide_string(valuename) do |name_ptr|
```

Comment:
Use `FFI::Pointer` here instead of `FFI::MemoryPointer`.  The helper is defined on the base class.

Reply:
Done